test/occupation.jl

```diff
@@ -134,3 +134,38 @@ if mpi_nprocs() == 1 # can't be bothered to convert the tests
     end
 end
 end
+
+if mpi_nprocs() == 1 # can't be bothered to convert the tests
+@testset "Fixed Fermi level" begin
+    testcase = magnesium
+
+    compute_scfres(fixed_εF=nothing) = begin
+        model = model_atomic(silicon.lattice, testcase.atoms, testcase.positions;
+                             temperature=0.01, smearing=DFTK.Smearing.FermiDirac(),
+                             fixed_εF, skip_fixed_εF_sane_check=true)
+        basis = PlaneWaveBasis(model; Ecut=5, kgrid=[2, 3, 4], fft_size=[15, 15, 15],
+                               kshift=[1, 0, 1]/2)
+        self_consistent_field(basis; n_bands=6)
+    end
+    scfres_ref = compute_scfres()
+    εF_ref = scfres_ref.εF
+    n_electrons_ref = scfres_ref.basis.model.n_electrons
+    @test n_electrons_ref == testcase.n_electrons
+
+    δεF = εF_ref / 4
+
+    for εF in [εF_ref - δεF, εF_ref + δεF]
+        scfres = compute_scfres(εF)
+        @test εF ≈ scfres.εF
+        op = εF > εF_ref ? :> : :<
+        @test eval(op)(DFTK.weighted_ksum(scfres.basis, sum.(scfres.occupation)), n_electrons_ref)
```

Comment:
That's pretty complicated for saving one line actually. I prefer that being written out twice.